### PR TITLE
Add sample for java functions framework

### PR DESCRIPTION
**Pull request by matthewrobertson**
Hello world sample that demonstrates using buildpacks to containerize a Functions Framework for Java app.

### CODE REVIEW

File: sample-functions-framework-java-mvn/pom.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.google</groupId>
+  <artifactId>sample-functions-java-mvn</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <properties>
+    <maven.compiler.target>11</maven.compiler.target>
```
Comment (jamesward) [inline sample-functions-framework-java-mvn/pom.xml:10]: Does this work with Java 1.8?

Comment (matthewrobertson) [reply sample-functions-framework-java-mvn/pom.xml:10]: I think 11 is the lowest version supported by the Functions Framework for Java

Comment (matthewrobertson) [reply sample-functions-framework-java-mvn/pom.xml:10]: and I don't see Java 8 listed in the docs here: https://cloud.google.com/functions/docs/concepts/exec#runtimes

Comment (jamesward) [reply sample-functions-framework-java-mvn/pom.xml:10]: Looks that way: https://github.com/GoogleCloudPlatform/functions-framework-java/blob/master/functions-framework-api/pom.xml#L65-L66

**Comment by jamesward** [comment]
One more thing... In our other Maven sample we include the Maven Wrapper so that users don't need to install Maven. Can you add that?

**Comment by jamesward** [comment]
Thanks for this!

**Comment by jamesward** [comment]
Oops. I should have tested this with `mvnw`. It failed I think due to a bug in the Buildpack:
```
=== Java - Maven (google.java.maven@0.9.0) ===
--------------------------------------------------------------------------------
Running "./mvnw clean package --batch-mode -DskipTests --quiet"
Done "./mvnw clean package --batch-mode -DskipTests --quiet" (11.33843831s)
=== Java - Functions Framework Invoker (google.java.functions-framework@1.0.1) ===
--------------------------------------------------------------------------------
Running "mvn --batch-mode dependency:copy-dependencies"
Done "mvn --batch-mode dependency:copy-dependencies" (676.682µs)
Failure: (ID: fce8c4c5) executing command "mvn --batch-mode dependency:copy-dependencies": exec: "mvn": executable file
 not found in $PATH
```
It used `mvnw` correctly, but then tried to use `mvn`.  I'll file a bug & pull out `mvnw` for now.

**Comment by jamesward** [comment]
For reference: https://github.com/GoogleCloudPlatform/buildpacks/issues/119

**Comment by jamesward** [comment]
Since that was fixed I reverted my commit that pulled out `mvnw` and it all seems to be working now.  Thanks again for adding this!

**Pull request merged**